nginx-fem-staging-redirects.conf

```diff
@@ -31,7 +31,7 @@ location ~* ^/about/(?:team|publications)/?$ {
 }
 
 # FEM Projects app routes for project index page (optional trailing slash)
-location ~* ^/projects/(?:[\w-]*?/)?[\w-]*?/[\w-]+?/?$ {
+location ~* "^/projects/(?:\w{2,4}?-?\w{0,4}/)?[\w-]*?/[\w-]+?/?$" {
```

Comment:
This regex is getting hard to read - I think we should simplify the setup and add another location block that handles the matching language code URL paths.

E.g. one for the normal project index routes (no language codes)
```
location ~* ^/projects/[\w-]*?/[\w-]+?/?$ {
```
and one that specifies the specific language codes in the paths
```
# note the change from \w to more restrictive posix classes for lang codes - i think they meet expectations.
location ~* "^/projects/[[:alpha:]]{2,4}(?:-[[:alnum:]]{0,4})?/[\w-]*?/[\w-]+?/?$" {
```

While the regex matching the language codes isn't simple it does allow more explicit matching on the paths with lang codes, e.g. 
`/projects/fr/bldigital/in-the-spotlight` and `projects/zh-cn/bldigital/in-the-spotlight/classify`

Reply:
Sounds good. Works for me when I test it locally, although RegExr doesn't seem to accept posix codes when I edit my test expression. https://regexr.com/6m2gu

Reply:
My only reservation is that this fails for usernames like `mr-bean` but we can deal with that if it comes up.

Reply:
> My only reservation is that this fails for usernames like `mr-bean` but we can deal with that if it comes up.

To avoid this we could shrink the list of supported language codes to what languages we know exist in our platform. Then match directly on them in a simpler regex vs the possible language code pattern space...? 
e.g. `^\/projects\/(?:(en|en-gb|pl|es)\/)?[\w-]*?\/[\w-]+?\/?$` and we expand that list as more languages are setup. I realise this requires some interventions by us but it might be worth it vs unexpected matching usernames...

Then we could iterate longer term to support the less well known codes / the available language code pattern space as required.

Reply:
That would work but the list of languages that we support is pretty long already. 
https://github.com/zooniverse/Panoptes-Front-End/tree/master/app/locales